atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -48,22 +48,25 @@ public String getUrl() {
         if (getServerDn().isPresent()) {
             return String.format("jdbc:oracle:thin:@(DESCRIPTION="
                             + "(ADDRESS=(PROTOCOL=%s)(HOST=%s)(PORT=%s))"
-                            + "(CONNECT_DATA=(SID=%s))"
+                            + "(CONNECT_DATA=(%s))"
                             + "(SECURITY=(SSL_SERVER_CERT_DN=\"%s\")))",
-                    getProtocol().getUrlString(), getHost(), getPort(), getSid(), getServerDn().get());
+                    getProtocol().getUrlString(), getHost(), getPort(), getConnectionDataString(), getServerDn().get());
         } else {
             return String.format("jdbc:oracle:thin:@(DESCRIPTION="
                             + "(ADDRESS=(PROTOCOL=%s)(HOST=%s)(PORT=%s))"
-                            + "(CONNECT_DATA=(SID=%s)))",
-                    getProtocol().getUrlString(), getHost(), getPort(), getSid());
+                            + "(CONNECT_DATA=(%s)))",
+                    getProtocol().getUrlString(), getHost(), getPort(), getConnectionDataString());
         }
     }
 
     @Override
     @Value.Derived
     @JsonIgnore
     public Optional<String> namespace() {
-        return Optional.of(getSid());
+        // If an SID is provided, maintain it - this is needed for legacy compatibility. But if a service name
+        // is provided, that identifies the database and we don't want to enforce consistency checks with the
+        // TimeLock client.
+        return getSid();
```

Comment:
somewhat confused, if you've specified `serviceName` you can no longer specify `sid` meaning that you get screwed over here whenever you call `namespace` i.e. you go from having a namespace to *not* having a namespace. What are the implications of this in atlas?

If anything it feels like `getConnectionDataString` should return the service name if it is specified, otherwise it returns the sid and that sid is *always* provided OR that if you specify serviceName, then you *need* to specify namespace explicitly when migrating over.

Let's chat about this face to face.

Reply:
In AtlasDB config, users can specify:

- a KVS-level `namespace` K (equals `keyspace` in Cassandra, `dbName` in Postgres, `sid` in Oracle)
- a timelock-level `client` T
- a global `namespace` G

All of them are optional. The current rules are

Non TimeLock users
- G is specified OR K is specified
- If both are specified G == K

TimeLock users
- G is specified OR (T and K are specified)
- If both are specified, G == T
- If both are specified, G == K

The interesting cases here are those where K is specified.

No TimeLock
- If G and K are both specified, then they must already be equal, and the behaviour of removing K is identical to the baseline.
- If G is not specified, then K doesn't matter as the timestamp bound is stored in the database being referenced.

TimeLock
- If G and K are specified, they are necessarily equal, and no change is needed.
- If T and K are specified while G is not, when migrating users must set the new value of G to be T (this is correct even if K was previously different from T), so that clients use the same namespace in TimeLock. Failure to do so will cause the service to not start.

We can discuss tomorrow. SID is always provided is not an option, as the deployment in question does not have that information as they do not operate the Oracle database in question. The second solution might be better if we decide it's needed, though it makes Oracle configs repetitive for TimeLock users.